lib/db/db.rb

```diff
@@ -9,26 +9,63 @@ def size
     all.size
   end
 
+  def all
+    reader.nil? ? [] : reader.run
+  end
+
+  def at(index)
+    empty || invalid(index) ? {} : all[index]
+  end
+
   def add(contact)
-    contacts = all
-    contacts << contact
+    contacts = all << format(contact)
     writer.run(contacts) if !writer.nil?
   end
 
-  def all
-    reader.nil? ? [] : reader.run
+  def update(contact)
+    contacts = updated_with(format(contact))
+    writer.run(contacts) if !writer.nil?
   end
 
-  def at(id)
-    all.empty? ? {} : all[id]
+  def search(term)
+    all.select { |contact| contact.any? {|key, val| val.to_s.include?(term)} }
   end
 
-  def search(term)
-    all.select{ |contact| contact.any? {|key, val| val.include?(term)} }
+  def index_of_id(id)
```

Comment:
That sound's like a leaking UI concern here, doesn't it?
I'm not saying that it's bad per say to have list, but maybe it makes the overall code cleaner to always use the ID as the identifier of a contact, instead of it's position in the list.